test/ex_doc/language/elixir_test.exs

```diff
@@ -460,6 +460,22 @@ defmodule ExDoc.Language.ElixirTest do
 
     warn(~m"`t:Calendar.date/9`")
 
+    assert warn(fn ->
+             assert autolink_doc(~m"[text](`fakefunction`)") == ~m"text"
+           end) =~ ~s[but it is invalid]
```

Comment:
could you include the reported reference in the warning string? I.e. this assertion should be:

```elixir
(...) =~ ~s["`fakefunction`" but it is invalid]
```

but I think we actually want:

```elixir
(...) =~ ~s["fakefunction" but it is invalid]
```

btw it's fine if we include more warning in the assertion, i.e.:

```elixir
(...) =~ ~s[documentation references "fakefunction" but it is invalid]
```

because we have some logic around including additional words like module, function, etc so stricter assertion can be helpful here.

Reply:
No problems! I've updated the code to not output the backticks on warning and improved the assertions :)